### PR TITLE
Leuta task7

### PR DESCRIPTION
**Pull request by denisleuta**
Description of the Titanic Data Analysis Code
This Python script performs an exploratory data analysis (EDA) on the Titanic dataset, utilizing several popular libraries such as Pandas, Seaborn, Matplotlib, and Plotly. The primary goal is to analyze passenger data to gain insights into survival rates, demographics, and other relevant factors. Below is a breakdown of the code's functionality:
1. Importing Libraries
The script begins by importing necessary libraries:
pandas for data manipulation and analysis.
seaborn and matplotlib.pyplot for data visualization.
plotly.express and plotly.graph_objects for interactive visualizations.
warnings to manage warning messages.
2. Loading Data
The code loads training and test datasets from CSV files located on the user's desktop. It combines these datasets into a single DataFrame for comprehensive analysis.
3. Data Preparation
Relevant columns (Survived, Pclass, Sex) are converted to categorical types to optimize analysis.
Descriptive statistics of the combined dataset are printed to provide an overview of the data.
The script counts the number of passengers in each class and calculates the average age of passengers grouped by class and sex.
4. Identifying Youngest and Oldest Passengers
The code identifies and prints the youngest and oldest passengers based on their class and sex.
5. Filtering Survivors
It filters out survivors whose names start with 'K' and sorts them by fare, printing the names of those who paid the highest and lowest fares.
6. Relatives Calculation
A new column is created to sum the number of relatives (siblings/spouses and parents/children) for each passenger. The maximum number of relatives among survivors is calculated and printed.
7. Fare Analysis
The average fare for passengers with cabins versus those without is calculated, along with their ratio.
8. Data Visualization
Several visualizations are created to represent different aspects of the data:
A count plot showing the distribution of passengers by class.
A Sankey diagram illustrating passenger flow by class, sex, and survival status.
A scatter plot visualizing the relationship between age and fare, considering class and number of relatives.
Histograms and count plots to analyze age distribution, passenger count by class, and gender distribution.
A pie chart depicting survival rates among passengers.
Box plots to compare fares across different classes.
A sunburst chart representing passenger distribution by class and gender.
A 3D scatter plot analyzing relationships between age, relatives, and fare.
9. Handling Missing Values
Missing values in the 'Fare' column are replaced with zero for further analysis.
10. Final Visualizations
The script concludes with additional scatter plots that explore relationships between age, fare, survival status, and gender.
Conclusion
This code provides a comprehensive analysis of Titanic passenger data through various statistical calculations and visualizations. It helps in understanding key factors that influenced survival rates during the tragic event. This description outlines the purpose and functionality of your code while providing context for each section's operations.

### CODE REVIEW

File: mypy.ini
Comment (Krekep) [inline mypy.ini:1]: Reason?

File: project/dataset/titanic.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+import plotly.graph_objects as go  # type: ignore
+import warnings
+
+
+# Load data from CSV files
+train = pd.read_csv(
+    "C:/Users/denis/OneDrive/Рабочий стол/Python/spbu-python-course-1/project/dataset/train.csv"
+)
+test = pd.read_csv(
+    "C:/Users/denis/OneDrive/Рабочий стол/Python/spbu-python-course-1/project/dataset/test.csv"
```
Comment (Krekep) [inline project/dataset/titanic.py]: Лучше использовать относительные пути

File: project/dataset/titanic.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt  # type: ignore
+import plotly.express as px  # type: ignore
+import plotly.graph_objects as go  # type: ignore
```
Comment (Krekep) [inline project/dataset/titanic.py:5]: А зачем здесь игнорирование типов

**Review by Krekep — CHANGES_REQUESTED** [review]
В целом хорошо, но необходимы исправления:
1. Вы не добавили используемые пакеты в requirements
2. Улучшите текстовое оформление ноутбука, к примеру стоит использовать заголовки, которые начинаются с знаков хэштега, для отделения смысловых частей задания друг от друга
3. Не обязательно, но упростит проверку --- рекомендую делать такие ноутбуки в облачных сервисах и прикладывать ссылку на них
4. Пишите ответ более явно, не кодом
5. Выведите `survivors_with_k_sorted`
6. Также требовалось найти не только макс/мин возраст, но и разницу между ними

File: mypy.ini
Comment (denisleuta) [reply mypy.ini:1]: Пытался исправить ошибки mypy. Сейчас удалил

File: project/dataset/titanic.py (hunk 1 above)
Comment (denisleuta) [reply project/dataset/titanic.py]: Заменил

File: project/dataset/titanic.py (hunk 2 above)
Comment (denisleuta) [reply project/dataset/titanic.py:5]: Без игнорирования выдает ошибки связанные с library stubs

**Review by Krekep — APPROVED** [review]
Зачтено

**Pull request merged**